src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts

```diff
@@ -50,37 +56,50 @@ export class VehicleTechnicalRecordComponent implements OnInit, AfterViewInit {
     return this.vehicleTechRecord?.vrms.filter(vrm => vrm.isPrimary === false);
   }
 
-  public get Roles() {
+  public get roles() {
     return Roles;
   }
 
   get vehicleTypes() {
     return VehicleTypes;
   }
 
-  handleSubmit(updateTechRecord: Function) {
-    this.handleFormState();
-    this.isInvalid ? updateTechRecord(false) : updateTechRecord(true)
-  }
-
-  handleFormState() {
-    const form = this.dynamicForm.sections.map(section => section.form);
-    this.isDirty = this.isAnyFormDirty(form);
-    this.isInvalid = this.isAnyFormInvalid(form);
-  }
-
   isAnyFormDirty(forms: Array<CustomFormGroup | CustomFormArray>) {
     return forms.some(form => form.dirty);
   }
 
   isAnyFormInvalid(forms: Array<CustomFormGroup | CustomFormArray>) {
     const errors: GlobalError[] = [];
-    forms.forEach(form => {
-      DynamicFormService.updateValidity(form, errors);
-    });
 
-    errors.length > 0 ? this.errorService.setErrors(errors) : this.errorService.clearErrors();
+    forms.forEach(form => DynamicFormService.updateValidity(form, errors));
+
+    errors.length > 0
+      ? this.errorService.setErrors(errors)
+      : this.errorService.clearErrors();
```

Comment:
Minor detail, but you can just do this
```suggestion
    errors.length
      ? this.errorService.setErrors(errors)
      : this.errorService.clearErrors();
```

Reply:
Done